packages/flame/lib/src/experimental/layout_component.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+
+enum Direction { horizontal, vertical }
+
+enum LayoutComponentAlignment { start, end, center }
+
+/// Super class for [RowComponent] and [ColumnComponent]
+/// A relayout is performed when
+///  - a change in this component's children takes place
+///  - the [gap] parameter is changed
+abstract class LayoutComponent extends PositionComponent {
+  LayoutComponent(
+    this.direction,
+    this.alignment,
+    this._gap,
+  );
+  final Direction direction;
+  final LayoutComponentAlignment alignment;
+  bool _allowSetSize = false;
+
+  /// gap between components
+  double _gap;
+
+  set gap(double gap) {
+    _gap = gap;
+    layoutChildren();
+  }
+
+  @override
+  set size(Vector2 size) {
+    assert(_allowSetSize, 'Setting the size is currently unsupported.');
+    this.size = size;
+  }
+
+  double get gap => _gap;
+
+  @override
+  void onChildrenChanged(Component child, ChildrenChangeType changeType) {
+    layoutChildren();
+  }
+
+  void layoutChildren() {
+    final list = children.whereType<PositionComponent>().toList();
+    if (list.isEmpty) {
+      return;
+    }
+    final currentPosition = Vector2.zero();
+    final vectorIndex = direction == Direction.horizontal ? 0 : 1;
+    final totalSizeOfComponents = list.fold<double>(
+      0,
+      (previousValue, element) => previousValue + element.size[vectorIndex],
+    );
+    _allowSetSize = true;
+    size[vectorIndex] = totalSizeOfComponents;
+    _allowSetSize = false;
```

Comment:
I don't think you need this variable -- can you just user `super.size`?

example:
```
class A {
  double size = 1.0;
}

class B extends A {
  @override
  set size(double size) {
    assert(false, 'Setting the size is currently unsupported.');
  }
  
  void foo() {
    // this.size = 2.0; -> this breaks
    super.size = 2.0;
  }
}

void main() {
  final b = B()..foo();
  print(b.size);
}
```